### ISSUE TITLE
Slow or No Start with No Network Connection

### ISSUE THREAD

**Issue by aj-foster** [FIRST-Tech-Challenge/scorekeeper#588]
**Describe the bug**

When starting the scoring system on a MacBook that does not have any active network connections (no LAN plugged in, and WiFi either disconnected or disabled), the system may take a very long time to start or fail to start altogether.

**To Reproduce**

1. Disconnect from all networks (unplug LAN and disconnect or disable WiFi).
2. Start the system "as normal", and enter admin password.
3. Observe there is no welcoming log message as usually occurs during startup, and browsers are not launched.
4. Visit `localhost` to see an error message.

**Expected behavior**

Expect the system to start up as normal, perhaps with the loopback IP displayed, and a requirement that the operator ensures the IP changes after reconnecting to a network.

**Screenshots**

**Device (please complete the following information):**
 - OS: macOS 10.15.7
 - Browser: N/A, doesn't get that far
 - Version: Tested with 4.1.5 and 4.1.6

**Additional context**

It's definitely unusual to use the software with no active network connection, but this happens during event setup when I need to confirm details of the event (like the list of teams), often in a location away from the router we use for scoring.

**Comment by tweirtx** [comment]
Strange, I'm not seeing this behavior on Linux or Windows. Wondering if it's a Mac specific thing

**Comment by jvens** [comment]
I was able to replicate this issue on Ubuntu 22.04.  If wifi is turned off and ethernet is disconnected it seems to hang during startup.  It is still hung up even if I turn wifi on while in that hung state.  I don't see anything particularly interesting in the log, but here it is anyway.
```
20:40:45.863 [main] [] INFO  org.eclipse.jetty.util.log - Logging initialized @748ms to org.eclipse.jetty.util.log.Slf4jLog
20:40:46.119 [main] [] INFO  org.usfirst.ftc.server.Server - Server boot id: 80c68416-9efb-4a67-9bbe-2aff1aebae55
20:40:46.314 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: SINGLE_SCORESHEET
20:40:46.341 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: API_MATCH
20:40:46.355 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: INSPECTION_FORM
20:40:46.355 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: INSPECTION_FORM
20:40:46.394 [main] [] INFO  org.usfirst.ftc.server.Server - Attempting to start server on port 80
20:40:46.395 [main] [] INFO  s.s.StaticFilesConfiguration - StaticResourceHandler configured with folder = /public
20:40:46.397 [main] [] INFO  s.s.StaticFilesConfiguration - External StaticResourceHandler configured with folder = ../uploads
20:40:46.496 [Thread-11] [] INFO  s.e.jetty.EmbeddedJettyServer - == Spark has ignited ...
20:40:46.497 [Thread-11] [] INFO  s.e.jetty.EmbeddedJettyServer - >> Listening on 0.0.0.0:80
20:40:46.499 [Thread-11] [] INFO  org.eclipse.jetty.server.Server - jetty-9.4.48.v20220622; built: 2022-06-21T20:42:25.880Z; git: 6b67c5719d1f4371b33655ff2d047d24e171e49a; jvm 11.0.17+8-post-Ubuntu-1ubuntu222.04
20:40:46.514 [Thread-11] [] INFO  org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
20:40:46.514 [Thread-11] [] INFO  org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
20:40:46.515 [Thread-11] [] INFO  org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
20:40:46.520 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@81be84f{/,null,AVAILABLE}
20:40:46.520 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@5403bd45{/apiv1,null,AVAILABLE}
20:40:46.520 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@5bc48b23{/api,null,AVAILABLE}
20:40:46.527 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@72299a23{/,null,AVAILABLE}
20:40:46.534 [Thread-11] [] INFO  o.e.jetty.server.AbstractConnector - Started ServerConnector@10f14014{HTTP/1.1, (http/1.1)}{0.0.0.0:80}
20:40:46.534 [Thread-11] [] INFO  org.eclipse.jetty.server.Server - Started @1422ms
20:40:46.534 [main] [] INFO  org.usfirst.ftc.server.Server - Server started on port 80
```

And going to localhost in the browser gives
```
HTTP ERROR 404 Not Found
URI:	/
STATUS:	404
MESSAGE:	Not Found
SERVLET:	org.eclipse.jetty.servlet.ServletHandler$Default404Servlet-1a570542
```

Starting the system with WiFi turned on gives the following log
```
20:47:04.832 [main] [] INFO  org.eclipse.jetty.util.log - Logging initialized @719ms to org.eclipse.jetty.util.log.Slf4jLog
20:47:05.043 [main] [] INFO  org.usfirst.ftc.server.Server - Server boot id: 8a1c79cf-3b0e-461e-8e27-60494ef45016
20:47:05.232 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: SINGLE_SCORESHEET
20:47:05.256 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: API_MATCH
20:47:05.269 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: INSPECTION_FORM
20:47:05.269 [main] [] INFO  o.f.ftc.common.utils.RouteMapping - Ordering dependent route registered: INSPECTION_FORM
20:47:05.308 [main] [] INFO  org.usfirst.ftc.server.Server - Attempting to start server on port 80
20:47:05.310 [main] [] INFO  s.s.StaticFilesConfiguration - StaticResourceHandler configured with folder = /public
20:47:05.313 [main] [] INFO  s.s.StaticFilesConfiguration - External StaticResourceHandler configured with folder = ../uploads
20:47:05.415 [Thread-11] [] INFO  s.e.jetty.EmbeddedJettyServer - == Spark has ignited ...
20:47:05.415 [Thread-11] [] INFO  s.e.jetty.EmbeddedJettyServer - >> Listening on 0.0.0.0:80
20:47:05.417 [Thread-11] [] INFO  org.eclipse.jetty.server.Server - jetty-9.4.48.v20220622; built: 2022-06-21T20:42:25.880Z; git: 6b67c5719d1f4371b33655ff2d047d24e171e49a; jvm 11.0.17+8-post-Ubuntu-1ubuntu222.04
20:47:05.433 [Thread-11] [] INFO  org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
20:47:05.433 [Thread-11] [] INFO  org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
20:47:05.434 [Thread-11] [] INFO  org.eclipse.jetty.server.session - node0 Scavenging every 660000ms
20:47:05.438 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@7b862fe5{/,null,AVAILABLE}
20:47:05.439 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@7583cae8{/apiv1,null,AVAILABLE}
20:47:05.439 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.h.ContextHandler@2d065d72{/api,null,AVAILABLE}
20:47:05.445 [Thread-11] [] INFO  o.e.j.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@75bb7b65{/,null,AVAILABLE}
20:47:05.452 [Thread-11] [] INFO  o.e.jetty.server.AbstractConnector - Started ServerConnector@115e7726{HTTP/1.1, (http/1.1)}{0.0.0.0:80}
20:47:05.453 [Thread-11] [] INFO  org.eclipse.jetty.server.Server - Started @1344ms
20:47:05.453 [main] [] INFO  org.usfirst.ftc.server.Server - Server started on port 80
20:47:05.490 [main] [] INFO  org.usfirst.ftc.server.Server - Unsupported Operating System: Linux, Browser not started
20:47:10.348 [qtp1068425436-32] [] INFO  org.usfirst.ftc.server.Server - Time: 11/11/2022 08:47:10:346 PM, Session: node04zmdtyy7jtocoeqqlpf0zm6w0, User: -, Route: http://localhost/ GET
20:47:10.546 [qtp1068425436-32] [] INFO  org.eclipse.jetty.server.RequestLog - [localhost] [0:0:0:0:0:0:0:1] - - [12/Nov/2022:02:47:10 +0000] "GET / HTTP/1.1" 200 6115 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36" 334000
20:47:10.590 [qtp1068425436-42] [] INFO  org.usfirst.ftc.server.Server - Time: 11/11/2022 08:47:10:590 PM, Session: node04zmdtyy7jtocoeqqlpf0zm6w0, User: -, Route: http://localhost/translations/en.js?version=1 GET
20:47:11.051 [qtp1068425436-41] [] INFO  org.eclipse.jetty.server.RequestLog - [localhost] [0:0:0:0:0:0:0:1] - - [12/Nov/2022:02:47:11 +0000] "GET /favicon-44dea7c1f52c129a8bee920a.ico HTTP/1.1" 200 848 "http://localhost/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36" 2000
```

**Comment by adwiii** [comment]
We have identified the issue and it will be fixed in the next release. Regardless of operating system, if there are no other network connections *at all*, the system will fail to start in the current release. 

@tweirtx, in my local Windows machine I do not hit this issue because Windows has the virtual ethernet adapter for WSL running regardless of other network connections, i.e. my Windows system will not hit this issue even when airplane mode is turned on.

**Comment by tweirtx** [comment]
Interesting, good to know

**Comment by pmichaud** [comment]
FWIW, I've hit this issue under Windows several times this season -- if no
network connection (IP address) is available, the software hangs on startup.

Pm

**Comment by maths22** [comment]
This has been fixed in v4.1.8
